### PR TITLE
Add difficulty feature

### LINKED ISSUE #166: Add difficulty feature
Add feature for users to rate the difficulty of a dish, from 0 to 5

### PR DESCRIPTION
**Pull request by AdarshChugani**
Add the `difficulty` feature

### CODE REVIEW

File: src/main/java/cookbuddy/model/recipe/attribute/Difficulty.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package cookbuddy.model.recipe.attribute;
+
+import static cookbuddy.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents the Difficulty of the recipe in the recipe book.
+ * Guarantees: immutable; rating is valid as declared in {@link #isValidDifficulty(int)}
```
Comment (kevinswk94) [inline src/main/java/cookbuddy/model/recipe/attribute/Difficulty.java]:
```suggestion
 * Guarantees: immutable; difficulty is valid as declared in {@link #isValidDifficulty(int)}
```

File: src/main/java/cookbuddy/model/recipe/attribute/Difficulty.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+     */
+    public Difficulty(int difficulty) {
+        requireNonNull(difficulty);
+        checkArgument(isValidDifficulty(difficulty), MESSAGE_CONSTRAINTS);
+        this.difficulty = difficulty;
+    }
+
+
+    /**
+     * Returns true if a given string is a valid serving size.
```
Comment (kevinswk94) [inline src/main/java/cookbuddy/model/recipe/attribute/Difficulty.java]:
```suggestion
     * Returns true if a given string is a valid difficulty.
```

File: src/main/java/cookbuddy/model/recipe/attribute/Difficulty.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * Guarantees: immutable; rating is valid as declared in {@link #isValidDifficulty(int)}
+ */
+public class Difficulty {
+
+    public static final String MESSAGE_CONSTRAINTS = "Difficulty should be an integer >= 0 and <= 5";
+
+    public final int difficulty;
+
+    /**
+     * Constructs a {@code Serving size}.
```
Comment (kevinswk94) [inline src/main/java/cookbuddy/model/recipe/attribute/Difficulty.java]:
```suggestion
     * Constructs a {@code Difficulty}.
```

**Review by kevinswk94 — CHANGES_REQUESTED** [review]
nice work, just some minor typos that you can batch commit from the suggestions directly.

**Review by kevinswk94 — APPROVED** [review]
LGTM

**Review by sharadhr — APPROVED** [review]
LGTM.

**Pull request merged**